dev/tasks/tasks.yml

```diff
@@ -941,6 +941,13 @@ tasks:
   test-r-devdocs:
     ci: github
     template: r/github.devdocs.yml
+    
+  test-r-depsource-auto:
+    ci: github
+    template: docker-tests/github.linux.yml
+    params:
+      flags: '-e ARROW_DEPENDENCY_SOURCE=AUTO -e ARROW_SOURCE_HOME="/arrow" -e LIBARROW_BUILD=TRUE'
+      image: ubuntu-r
```

Comment:
This isn't the image you want to use. Look at the build logs: it's not using the build script we ship with the R package, and it builds a shared library. So it's not exercising any of your new code.

Pick one of the images enumerated below this and add a job like it that has the `flags` provided. That will exercise `r/inst/build_arrow_static.sh`.

Reply:
Sorry, I don't understand what you mean by "has the `flags` provided"?

Reply:
```
  test-r-depsource-auto:
    ci: azure
    template: r/azure.linux.yml
    params:
      r_org: rocker
      r_image: r-base
      r_tag: latest
      flags: '-e ARROW_DEPENDENCY_SOURCE=AUTO'
```

Then it looks like you'll need to add `{{ flags|default("") }}` around L53 in dev/tasks/r/azure.linux.yml so that the job picks them up since that parameter isn't defined in that template yet. (I think there's some refactoring we can do here to clean up these templates, but that's for another time.)